pkg/provider/aws/resources/securitygroup.go

```diff
@@ -77,9 +81,14 @@ func (sg *SecurityGroup) KlothoConstructRef() []core.AnnotationKey {
 
 // Id returns the id of the cloud resource
 func (sg *SecurityGroup) Id() core.ResourceId {
-	return core.ResourceId{
+	id := core.ResourceId{
 		Provider: AWS_PROVIDER,
 		Type:     SG_TYPE,
 		Name:     sg.Name,
 	}
+	if sg.Vpc != nil {
+		// Realistically, this should only be the case for tests
+		id.Namespace = sg.Vpc.Name
+	}
```

Comment:
In all compiles, `sg.Vpc` is never nil, but in various unit tests they leave it nil. Instead of tracking down all the unit tests and adding in the VPC just to make the security group insertable, I opted for making it technically optional (in terms of the id function).